app/models/claim/advocate_interim_claim.rb

```diff
@@ -65,6 +69,22 @@ def requires_case_type?
       false
     end
 
+    def agfs?
```

Comment:
⚠️  The introduction of these methods (`agfs?`, `final?` and `interim?`) caused all kind of side effects, because some checks in the code were, for example, using only `interim?` to check for a LGFS interim.

For `agfs?` ones, there's quite a lot of checks on this but I'm not quite sure if they're necessary to be changed to ensure they're `agfs? && !interim?` 😰  (I'm expecting these ones to be less problematic as the new type of claim has a brand new flow, so any unexpected behaviour will be fixed when we come to make the changes required and realise its not returning or showing what we expect it to be.

Also, initially, for consistency purposes, I've introduced the method `final?` to the existent `AdvocateClaim` but that was triggered a bunch of unexpected behaviours and failing validations on things like fees. I've decided to revert that change as I don't own enough knowledge right now to be able to make changes to the existent code and ensure the existent behaviour was preserved.